src/services/platform.service.js

```diff
@@ -92,24 +93,54 @@ export const initializePath = () => {
           return resolve();
         }
 
-        // For users with a standard Node installation, node will be in
-        // /usr/local/bin
-        // For users using NVM, the path to Node will be added to `.bashrc`.
-        // Add both to the PATH.
-        try {
-          childProcess.exec(
-            'source ~/.bashrc && echo $PATH',
-            (err, updatedPath) => {
-              if (updatedPath) {
-                window.process.env.PATH = `/usr/local/bin:${updatedPath}`;
-              }
-
-              resolve();
+        // Get login username
+        const username = os.userInfo().username;
+
+        childProcess.exec(
+          `dscl . -read /Users/${username} UserShell`,
+          (error, stdout) => {
+            if (error) {
+              resolve(error);
             }
-          );
-        } catch (e) {
-          resolve(e);
-        }
+
+            // Check which shell is in use e.g. /bin/bash or /bin/zsh
+            // Get the name of the shell after the last instant of '/'
+            const shell = stdout
+              .substring(stdout.lastIndexOf('/') + 1, stdout.length)
+              .trim();
+
+            // Format the config file using correct shell
+            let sourceFile =
+              shell === 'fish' ? '.config/fish/config.fish' : `.${shell}rc`;
+
+            // Check the file actually exists first
+            const fileExists = fs.existsSync(`${os.homedir()}/${sourceFile}`);
+
+            // If file doesn't exist then source .bash_profile instead
+            if (!fileExists) {
+              sourceFile = '~/.bash_profile';
```

Comment:
Should this just be `.bash_profile`? The `~/` is prepended below.

```suggestion
              sourceFile = '.bash_profile';
```

Reply:
Whoops, forgot that one 😅 Thanks!